lib/Net/Async/Blockchain/Client/Websocket.pm

```diff
@@ -150,24 +154,44 @@ L<Ryu::Source>
 sub _request {
     my ($self, $method, @params) = @_;
 
-    my $obj = {
+    my $url = URI->new($self->endpoint);
+
+    # this is a simple block number request
+    my $timer_call = {
+        id     => 1,
+        method => 'eth_blockNumber',
+        params => []};
+
+    # we need to keep sending requests to the node
+    # otherwise after some period of time we just
+    # get disconnected by the peer, 5 seconds is enough
+    # to keep the connection alive.
+    my $timer = IO::Async::Timer::Periodic->new(
+        interval => 5,
+        on_tick  => sub {
+            $self->websocket_client->send_text_frame(encode_json_utf8($timer_call));
```

Comment:
Would a simpler [`send_ping_frame()`](https://metacpan.org/pod/Net::Async::WebSocket::Protocol#send_TYPE_frame) work as well?

Reply:
I tried using it but the node seems to not receive the request when using the type ping on it, so I will keep as just text and actually is just a valid text command that we want to send anyway.